version: "3.9"

volumes:
  pg_altiora_data:

services:
  postgres:
    container_name: altiora_database
    image: postgres:17.1-alpine3.20
    env_file: .env
    volumes:
      - pg_altiora_data:/var/lib/postgresql/data
    ports:
      - 5433:5432
    healthcheck:
      test: pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}
      interval: 5s
      timeout: 10s
      retries: 5

  rabbitmq:
    image: rabbitmq:3-management
    container_name: altiora_rabbitmq
    ports:
      - "5672:5672"
      - "15672:15672"
    environment:
      RABBITMQ_DEFAULT_USER: user
      RABBITMQ_DEFAULT_PASS: pass

  celery:
    build:
      context: ./backend
    container_name: altiora_celery
    command: celery -A altiora_backend worker --loglevel=info
    volumes:
      - ./backend:/app
    env_file: .env
    working_dir: /app
    depends_on:
      - postgres
      - rabbitmq
